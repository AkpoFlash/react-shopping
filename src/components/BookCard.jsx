import React from 'react';
import { Card, Icon, Image, Button } from 'semantic-ui-react';

const BookCard = (props) => {
  const handleAddBookToCard = (e) => {
    props.addBookToCard(props);
  }
  return (
    <Card>
      <Image src={ props.image } />
      <Card.Content>
        <Card.Header>
          { props.title }
        </Card.Header>
        <Card.Meta>
          <span className='date'>
            { props.author }
          </span>
        </Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <span>
          { props.price }
          <Icon name='rub' />
        </span>
      </Card.Content>
      <Button onClick={ handleAddBookToCard }>Добавить в корзину { props.addedCount > 0 ? `(${ props.addedCount })`: '' }</Button>
    </Card>
  );
};

export default BookCard;
