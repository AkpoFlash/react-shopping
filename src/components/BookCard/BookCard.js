import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { addBookToCard, removeBookFromCard } from '../../actions/cards';
import { t } from '../../helpers';
import { COLOR_WHITE, COLOR_BLACK, COLOR_GRAY } from '../../constants/styles';

const StyledBookCard = styled.div`
	width: 170px;
	border: 1px solid ${ COLOR_GRAY };
	border-radius: 10px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
`;

const Image = styled.img`
	width: 100%;
	height: 200px;
	margin: 15px 0;
	object-fit: contain;
`;

const Content = styled.div`
	padding: 5px 10px;
	display: flex;
	flex-direction: column;
	border-top: 1px solid ${ COLOR_GRAY };
`;

const ContentTitle = styled.h4`
	font-weight: 700;
  font-size: 1rem;
  display: inline-block;
  margin: 0;
  word-wrap: break-word;
`;

const Price = styled.span`
  svg {
    width: 20px;
    height: 20px;
    display: inline-block;
    background: ${ COLOR_BLACK };
  }
`;

const Button = styled.button`
	width: 100%;
	height: 35px;
	background: ${ COLOR_GRAY };
  color: ${ COLOR_WHITE };
  cursor: pointer;
`;

const BookCard = (props) => {
  
  const handleAddBookToCard = (e) => {
    props.addBookToCard(props);
  }

  return (
    <StyledBookCard>
      <Image src={ props.image } />
      <Content>
        <ContentTitle>
          { props.title }
        </ContentTitle>
        <div>
          <span className='date'>
            { props.author }
          </span>
        </div>
      </Content>
      <Content>
        <Price>
          { props.price }
          <svg>

          </svg>
        </Price>
      </Content>
      <Button onClick={ handleAddBookToCard }>
        {t('Add to basket')} { props.addedCount > 0 ? `(${ props.addedCount })`: '' }
      </Button>
    </StyledBookCard>
  );
};

const mapStateToProps = (state, props) => ({
  addedCount: state.card.items.reduce( (count, book) => count + (book.id === props.id ? 1 : 0), 0),
});

const mapDispatchToProps = (dispatch) => ({
  addBookToCard: book => dispatch(addBookToCard(book)),
  removeBookFromCard: book => dispatch(removeBookFromCard(book)),
})

export default connect(mapStateToProps, mapDispatchToProps)(BookCard);
