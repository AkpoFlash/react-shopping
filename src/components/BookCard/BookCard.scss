.book-card{
	width: 170px;
	border: 1px solid var(--secendaryColor);
	border-radius: 10px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
}

.book-card__content{
	padding: 5px 10px;
	display: flex;
	flex-direction: column;
	border-top: 1px solid var(--secendaryColor);
}

.book-card__image{
	width: 100%;
	height: 200px;
	margin: 15px 0;
	object-fit: contain;
}

.book-card header{
	font-weight: 700;
	font-size: 1.25rem;
}

.book-card__currency{
	width: 20px;
	height: 20px;
	display: inline-block;
	background: #000;
}

.book-card__button{
	width: 100%;
	height: 35px;
	background: var(--secendaryColor);
	color: #fff;
}