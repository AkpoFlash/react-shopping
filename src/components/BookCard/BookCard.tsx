import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import * as PropTypes from 'prop-types';

import { addBookToCard, removeBookFromCard } from '~/actions/cards';
import { t } from '~/helpers';
import { COLOR_WHITE, COLOR_BLACK, COLOR_GRAY, COLOR_TEXT } from '~/constants/styles';

const StyledBookCard = styled.div`
  height: 100%;
	width: 100%;
	border: 1px solid ${ COLOR_GRAY };
	border-radius: 10px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
`;

const Image = styled.img`
	width: 100%;
	height: 200px;
	margin: 15px 0;
	object-fit: contain;
`;

const Content = styled.div`
	padding: 5px 10px;
	display: flex;
	flex-direction: column;
	border-top: 1px solid ${ COLOR_GRAY };
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: ${ COLOR_TEXT };
`;

const ContentTitle = styled.h4`
	font-weight: 700;
  font-size: 1rem;
  display: inline-block;
  margin: 0;
  word-wrap: break-word;
`;

const Price = styled.span`
  svg {
    width: 20px;
    height: 20px;
    display: inline-block;
    background: ${ COLOR_BLACK };
  }
`;

const Button = styled.button`
	width: 100%;
	height: 35px;
	background: ${ COLOR_GRAY };
  color: ${ COLOR_WHITE };
  cursor: pointer;
`;

export const BookCard: React.FunctionComponent<any> = (props) => {
  const { id, title, author, price, image } = props;
  
  const handleAddBookToCard = (e) => {
    props.addBookToCard(props);
  }

  return (
    <StyledBookCard>
      <StyledLink to={`/books/${ id }`} >
        <Image src={ image } />
        <Content>
          <ContentTitle>
            { title }
          </ContentTitle>
          <div>
            <span className='date'>
              { author }
            </span>
          </div>
        </Content>
        <Content>
          <Price>
            { price }
            <svg>

            </svg>
          </Price>
        </Content>
      </StyledLink>
      <Button onClick={ handleAddBookToCard }>
        {t('Add to basket')} { props.addedCount > 0 ? `(${ props.addedCount })`: '' }
      </Button>
    </StyledBookCard>
  );
};

const mapStateToProps = ({ card, languages }, props) => ({
  addedCount: card.items.reduce( (count, book) => count + (book.id === props.id ? 1 : 0), 0),
  usersLang: languages.usersLang,
});

const mapDispatchToProps = (dispatch) => ({
  addBookToCard: book => dispatch(addBookToCard(book)),
  removeBookFromCard: book => dispatch(removeBookFromCard(book)),
})

BookCard.propTypes = {
  addedCount: PropTypes.number.isRequired,
  usersLang: PropTypes.string.isRequired,
  addBookToCard: PropTypes.func.isRequired,
  removeBookFromCard: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(BookCard));
