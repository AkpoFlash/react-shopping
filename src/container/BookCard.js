import { connect } from 'react-redux';
import BookCard from '../components/BookCard';
import { addBookToCard, removeBookFromCard } from '../actions/cards';

const mapStateToProps = (state, props) => ({
  addedCount: state.card.items.reduce( (count, book) => count + (book.id === props.id ? 1 : 0), 0),
});

const mapDispatchToProps = (dispatch) => ({
  addBookToCard: book => dispatch(addBookToCard(book)),
  removeBookFromCard: book => dispatch(removeBookFromCard(book)),
})

export default connect(mapStateToProps, mapDispatchToProps)(BookCard);
