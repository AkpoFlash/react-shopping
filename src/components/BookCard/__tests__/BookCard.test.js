import React from 'react';

import { BookCard } from '../BookCard';

describe('Dumb BookCard', () => {
	const bookCard = shallow(<BookCard />);

	it('check BookCard snapshot', () => {
		expect(bookCard).toMatchSnapshot();
	});
});