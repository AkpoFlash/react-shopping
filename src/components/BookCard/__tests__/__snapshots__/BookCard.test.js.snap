// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dumb BookCard check BookCard snapshot 1`] = `
<styled.div>
  <Styled(Link)
    to="/books/0"
  >
    <styled.img
      src="https://cv7.litres.ru/sbc/33231270_cover_185-elektronnaya-kniga-den-braun-proishozhdenie-27624091.jpg"
    />
    <styled.div>
      <styled.h4>
        Происхождение
      </styled.h4>
      <div>
        <span
          className="date"
        >
          Дэн Браун
        </span>
      </div>
    </styled.div>
    <styled.div>
      <styled.span>
        710
        <svg />
      </styled.span>
    </styled.div>
  </Styled(Link)>
  <styled.button
    onClick={[Function]}
  >
    Add to basket
     
  </styled.button>
</styled.div>
`;
